packages/webapp/src/components/SensorReadingsLineChart/index.jsx

```diff
@@ -61,11 +61,8 @@ const PureSensorReadingsLineChart = ({
 
   const handleLegendClick = (entry) => {
     setLegendsList((legends) => {
-      const activeCount = Object.values(legends).filter((l) => l.isActive).length;
       const isActive = legends[entry.value].isActive;
-      if ((activeCount > 1 && isActive) || (activeCount >= 1 && !isActive)) {
-        legends[entry.value].isActive = !isActive;
-      }
+      legends[entry.value].isActive = !isActive;
```

Comment:
Use the produce function from immer if any function needs to update function arguments.
RTK uses immer internally but react setState doesn't

Reply:
I have added this in since it is well used throughout the app. But I can admit I haven't quite figured out the value of immer. Is it basically just more efficient or something because it doesn't overwrite the entire state on each setstate call?

Reply:
useState is supposed to be immutable. When setState is called, the previous state reference will point to a new state.
const [counter, setCounter] = useState(0);
setCounter(1)
console.log(counter)
Should log 0.
counter will point to the next value 1 on the next render.

People say setState behavior will be unpredictable when state is mutated.
I haven't had issues with setState(prev => next). I just blindly follow the immutable pattern. 
With immer, you can trust that it won't mutate the argument and the result state won't be mutated.

I haven't been up to date with react. It seems they moved onto useImmer
https://beta.reactjs.org/learn/adding-interactivity

Reply:
OK thanks for the extra info!